compiler/utils.go

```diff
@@ -738,3 +740,55 @@ func (st signatureTypes) Param(i int, ellipsis bool) types.Type {
 func ErrorAt(err error, fset *token.FileSet, pos token.Pos) error {
 	return fmt.Errorf("%s: %w", fset.Position(pos), err)
 }
+
+// FatalError is an error compiler panics with when it encountered a fatal error.
+//
+// FatalError implements io.Writer, which can be used to record any free-form
+// debugging details for human consumption. This information will be included
+// into String() result along with the rest.
+type FatalError struct {
+	cause interface{}
+	stack []byte
+	clues bytes.Buffer
```

Comment:
Consider using `strings.Builder` here since it might be a better fit.

Reply:
Done.